src/tls/workload.rs

```diff
@@ -146,12 +150,14 @@ where
         let mut acceptor = self.provider.clone();
         Box::pin(async move {
             let tls = acceptor.fetch_cert(&conn).await?;
-            tokio_rustls::TlsAcceptor::from(tls)
+            let tls_accept = tokio_rustls::TlsAcceptor::from(tls)
                 .accept(conn)
-                .map_err(TlsError::Handshake)
-                .await
-        })
-    }
+                .map_err(TlsError::Handshake);
+            timeout(Duration::from_secs(TLS_HANDSHAKE_TIMEOUT), tls_accept).map_err(move |e| {
```

Comment:
is https://docs.rs/tls-listener/latest/tls_listener/struct.Builder.html#method.handshake_timeout not sufficient here? defaults to 10s already

Reply:
Ah, I missed that, yep - should handle it, reverted that commit.